ui-tests/searchProduct.spec.js

```diff
@@ -0,0 +1,229 @@
+    quantity: "101",
+    shipping: "1",
+    category: sampleCategory._id,
+    price: "10",
+  });
+
+  const mechKeyboardProduct = await productModel.create({
+    name: "Mechanical keyboard with RGB light",
+    slug: "Mechanical-keyboard-with-RGB-light",
+    description: "RGB mechanical keyboard",
+    quantity: "123",
+    shipping: "0",
+    category: sampleCategory._id,
+    price: "91",
+  });
+
+  // Push product ids to this array, to keep track which products to delete later
+  createdProductIds.push(wirelessMouseProduct._id);
+  createdProductIds.push(mechKeyboardProduct._id);
+
+  const imageBuffer = fs.readFileSync("test-images/toycar.jpeg");
+  const productsData = Array.from({ length: 8 }, (_, i) => ({
+    name: `Electronic toy car ${i + 1}`,
+    slug: `Electronic-toy-car-${i + 1}`,
+    description: `Fast and durable electronic toy car ${i + 1}`,
+    quantity: `${i + 1}`,
+    shipping: "1",
+    category: sampleCategory._id,
+    price: `${i + 1}`,
+    photo: {
+      data: imageBuffer,
+      contentType: "image/jpeg",
+    },
+  }));
+
+  // Create more products for test
+  for (const product of productsData) {
+    const createdProduct = await productModel.create(product);
+    createdProductIds.push(createdProduct._id);
+  }
+});
+
+test.afterAll(async () => {
+  // Delete all products created in this test
+  await productModel.deleteMany({ _id: { $in: createdProductIds } });
+
+  // Delete the category created in this test
+  await categoryModel.deleteMany({
+    name: "IT accessories and other electronics",
+  });
+
+  // Delete the user created in this test
+  await userModel.deleteMany({ email: "sammyrae@mail.com" });
+
+  // Close the connection with Mongo DB since the test has finished running
+  await mongoose.disconnect();
+});
+
+// Helper function to be used in the two tests below
+async function searchAndLoadmoreVerification(page) {
+  // Click on the search link on the nav bar
+  await page.getByRole("link", { name: "Search" }).click();
+
+  // Expect that the search related text should be present
+  await expect(
+    page.getByRole("textbox", { name: "Search for a product" })
+  ).toBeVisible();
+  await expect(
+    page.getByRole("heading", { name: "Search Results" })
+  ).toBeVisible();
+
+  // Fill the search input with the text
+  await page
+    .getByRole("textbox", { name: "Search for a product" })
+    .fill("Electronic");
+  await page.getByRole("button", { name: "Search" }).click();
+
+  // Expect 6 products to be found
+  await expect(page.getByRole("heading", { name: "Found 6" })).toBeVisible();
+
+  // Expect the important product values to be present
+  await expect(
+    page.getByRole("heading", { name: "Electronic toy car 8" })
+  ).toBeVisible();
+  await expect(page.getByText("$ 8")).toBeVisible();
+
+  // Assert buttons like "add to cart" and "product details" are present
+  await expect(page.locator(".card-body > button").first()).toBeVisible();
```

Comment:
can we something less brittle here like using queryByRole and using name: Add To Cart instead?

Reply:
```
page.getByRole("button", { name: "Add To Cart" }).first();
```

Reply:
something like that

Reply:
Ah okay, will try this instead